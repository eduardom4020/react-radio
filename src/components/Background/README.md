# Background
>This component is a view that can have it's background colored. Should be used as wrapper.

### Props
These are the props accepted by the component, pay attention to the description, inputs and behaviour:

---

##### color *(string)*
*The background color in wich the wrapper will be filled. You should pass to it any color code accepted by CSS, like hexadecial, rgb or the color name.*
**example**: <Background color='#ff0000' />, *this will paint background in red*. 

---