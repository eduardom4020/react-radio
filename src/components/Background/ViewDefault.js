import styled from 'styled-components';
import { View, StatusBar } from 'react-native';

const BackgroundViewDefault = styled(View)`
    width: 100%;
    height: 100%;
    background-color: ${props => props.color || 'white'};
    flex-direction: column;
    justify-content: space-between;
    align-items: center;
    padding: ${props => props.statusBarHeight || StatusBar.currentHeight || 0}px 0 0 0;
`;

export default BackgroundViewDefault;