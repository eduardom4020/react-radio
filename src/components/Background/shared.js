import { css } from 'styled-components';

export const BackgroundStyle = css`
    width: 100%;
    height: 100%;
    background-color: ${props => props.color || 'white'};
    flex-direction: column;
    justify-content: space-between;
    align-items: center;
`;