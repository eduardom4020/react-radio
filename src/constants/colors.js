export const BLACK_LIKE = '#20202b';
export const DARK_GREY = '#2e2e3a';
export const MEDIUM_GREY = '#a3acbe';
export const MEDIUM_ORANGE = '#eeae61';
export const METALIC_BLACK = '#191313';