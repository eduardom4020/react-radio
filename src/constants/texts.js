export const APP_TITLE = 'STATIONS';