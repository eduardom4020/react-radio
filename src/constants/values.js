export const DEFAULT_IMAGE_BT_SIZE = 192;
export const DEFAULT_ICON_BT_SIZE = 24;
export const DEFAULT_RADIO_PLAYING = 'CURRENTLY PLAYING'