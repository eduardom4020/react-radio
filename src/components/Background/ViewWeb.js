import React from 'react';
import styled, { createGlobalStyle } from 'styled-components';
import { DARK_GREY, BLACK_LIKE } from '~/src/constants/colors';
import { BackgroundStyle } from './shared';

const Body = createGlobalStyle`
    body {
        width: 100%;
        height: 100%;
        height: -webkit-fill-available;
        background-color: ${props => props.color || BLACK_LIKE};
        display: flex;
        flex-direction: column;
        justify-content: center;
        align-items: center;
    }
`;

const StyledWrapper = styled.div`${BackgroundStyle}`;
const BackgroundWrapper = styled(StyledWrapper)`
    width: 35vw;
    height: 80vh;
    border: 1px solid white;
    border-radius: 64px;
`;

const BackgroundWebView = props => {
    const { children, ...extraProps } = props;
    console.log('On background web view')
    return (
        <>
            <Body color={BLACK_LIKE} />
            <BackgroundWrapper {...extraProps} >
                { children }
            </BackgroundWrapper>
        </>
    );
};

export default BackgroundWebView;