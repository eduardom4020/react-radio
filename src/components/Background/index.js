import View from './ViewDefault';

import ViewWeb from './ViewWeb';
export const BackgroundWebView = ViewWeb;

export default View;