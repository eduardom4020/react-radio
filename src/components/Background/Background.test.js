import React from 'react';
import { Text, View } from 'react-native';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { DARK_GREY } from '~/src/constants/colors';
import Background from '~/src/components/Background';

describe('Background', () => {
    test('Render?', () => {
        const component = shallow(
            <Background />
        );

        expect(toJson(
            component, {
                noKey: false,
                mode: 'deep'
            }
        )).toMatchSnapshot();
    });

    test('Render with custom color?', () => {
        const component = shallow(
            <Background color={DARK_GREY} />
        );

        expect(toJson(
            component, {
                noKey: false,
                mode: 'deep'
            }
        )).toMatchSnapshot();
    });
    
    test('Render with children?', () => {
        const component = shallow(
            <Background color={DARK_GREY} >
                <View><Text>Test1</Text></View>
                <Text>Test2</Text>
            </Background>
        );

        expect(toJson(
            component, {
                noKey: false,
                mode: 'deep'
            }
        )).toMatchSnapshot();
    });
});