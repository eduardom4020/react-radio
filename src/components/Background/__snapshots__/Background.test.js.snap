// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Background Render with children? 1`] = `
<StyledComponent
  color="#2e2e3a"
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
    width: 100%;
    height: 100%;
    background-color: ",
          [Function],
          ";
    flex-direction: column;
    justify-content: space-between;
    align-items: center;
",
        ],
      },
      "displayName": "Styled(View)",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": [Function],
      "toString": [Function],
      "usesTheme": false,
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
>
  <View>
    <Text>
      Test1
    </Text>
  </View>
  <Text>
    Test2
  </Text>
</StyledComponent>
`;

exports[`Background Render with custom color? 1`] = `
<StyledComponent
  color="#2e2e3a"
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
    width: 100%;
    height: 100%;
    background-color: ",
          [Function],
          ";
    flex-direction: column;
    justify-content: space-between;
    align-items: center;
",
        ],
      },
      "displayName": "Styled(View)",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": [Function],
      "toString": [Function],
      "usesTheme": false,
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;

exports[`Background Render? 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
    width: 100%;
    height: 100%;
    background-color: ",
          [Function],
          ";
    flex-direction: column;
    justify-content: space-between;
    align-items: center;
",
        ],
      },
      "displayName": "Styled(View)",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": [Function],
      "toString": [Function],
      "usesTheme": false,
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;
